File: hospital-appointment-frontend/src/api/hospital/medicalRecord.ts
/**
 * 门诊病历 API
 */

import request from '../request'

export interface MedicalRecord {
  recordID: number
  visitID: number
  appointmentID: string
  patientID: number
  doctorID: number
  doctorName?: string
  patientName?: string
  departmentName?: string
  chiefComplaint?: string
  presentIllness?: string
  pastHistory?: string
  allergyHistory?: string
  physicalExamination?: string
  auxiliaryExamination?: string
  preliminaryDiagnosis?: string
  finalDiagnosis?: string
  treatmentPlan?: string
  medicalAdvice?: string
  remarks?: string
  createTime: string
  updateTime?: string
}

export interface Prescription {
  prescriptionID: number
  recordID: number
  visitID: number
  patientID: number
  doctorID: number
  medicineName: string
  medicineSpec?: string
  dosage?: string
  usage?: string
  frequency?: string
  course?: string
  quantity?: string
  unit?: string
  price?: number
  totalPrice?: number
  remarks?: string
  prescriptionType?: number
  status?: number
  createTime?: string
}

export interface RecordDetail {
  record: MedicalRecord
  prescriptions: Prescription[]
}

export const medicalRecordAPI = {
  /**
   * 获取患者病历列表
   */
  getByPatient: (patientId: number) => {
    return request.get<MedicalRecord[]>(`/hospital/medical-record/patient/${patientId}`)
  },

  /**
   * 获取病历详情（含处方）
   */
  getDetail: (recordId: number) => {
    return request.get<RecordDetail>(`/hospital/medical-record/${recordId}`)
  },

  /**
   * 保存病历
   */
  save: (data: Partial<MedicalRecord>) => {
    return request.post<MedicalRecord>('/hospital/medical-record', data)
  }
}

File: hospital-appointment-backend/src/main/java/com/hospital/appointment/entity/MedicalRecord.java
package com.hospital.appointment.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 门诊病历实体类
 * 新增
 */
@Data
@TableName("medical_record")
public class MedicalRecord implements Serializable {
    
    @TableId(type = IdType.AUTO)
    private Long recordID;
    
    /** 就诊记录ID */
    private Long visitID;
    
    /** 挂号记录ID */
    private String appointmentID;
    
    /** 患者ID */
    private Long patientID;
    
    /** 医生ID */
    private Long doctorID;
    
    /** 主诉 */
    private String chiefComplaint;
    
    /** 现病史 */
    private String presentIllness;
    
    /** 既往史 */
    private String pastHistory;
    
    /** 过敏史 */
    private String allergyHistory;
    
    /** 体格检查 */
    private String physicalExamination;
    
    /** 辅助检查 */
    private String auxiliaryExamination;
    
    /** 初步诊断 */
    private String preliminaryDiagnosis;
    
    /** 最终诊断 */
    private String finalDiagnosis;
    
    /** 治疗方案 */
    private String treatmentPlan;
    
    /** 医嘱 */
    private String medicalAdvice;
    
    /** 备注 */
    private String remarks;
    
    /** 创建时间 */
    private LocalDateTime createTime;
    
    /** 更新时间 */
    private LocalDateTime updateTime;
    
    /** 非数据库字段：医生姓名 */
    @TableField(exist = false)
    private String doctorName;
    
    /** 非数据库字段：患者姓名 */
    @TableField(exist = false)
    private String patientName;

    /** 非数据库字段：科室名称 */
    @TableField(exist = false)
    private String departmentName;
}
